### ISSUE TITLE
Menu Link: [wblink] not working

### ISSUE THREAD

**Issue by instantflorian** [WBCE/WBCE_CMS#462]
When entering a wblink shortcut (eg [wblink2] ) in the menu link > external link field, the links is not generated. In former versions of WBCE this was possible and this feature should work again, since this was the best way to build a menu link to a specific manually added anchor (not: automatically generated section anchors) on the site. 
![2019-11-14_103152](https://user-images.githubusercontent.com/1433754/68844447-15f8aa80-06ca-11ea-8b97-8493ce5cc732.jpg)
![2019-11-14_103142](https://user-images.githubusercontent.com/1433754/68844448-15f8aa80-06ca-11ea-970f-053600087998.jpg)

**Comment by WebDesignWorx** [comment]
Hello Florian, 
the issue is with the conversion of [wblinkXX] entries only, or other entries as well?

**Comment by instantflorian** [comment]
Only with [wblinkXX].
The error appears only on websites without the line
define('SM2_CORRECT_MENU_LINKS',true);
in the config.php.

**Comment by instantflorian** [comment]
Fixed with 9bcd21754937fbbb3d0a359f34d2b68a100935b1

**Comment by mrbaseman** [comment]
@WebDesignWorx One thing which I didn't understand yet: why doesn't the Output Filter work in this case? Is the output of this code inserted too late? If so, we might have a more general problem with the order in which things are processed.

**Comment by WebDesignWorx** [comment]
@mrbaseman 
I think it's because nothing is being ouput in this case, so the outputfilter doesn't get hold of the [wblinkXX] token.
If you would echo the variable just at the same line in the code, the ouputfilter would work just fine.
I had a similar WTF moment with it until I realized.

Anyways, the code I implemented yesterday could be rewritten using preg_match/RegEx if you are so inclined. 

Best regards, 
Christian
